sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,378 @@
+
+The reason for this limitation is that currently, no backend supported by
+{dpcpp} supports exporting memory with a non-linear layout. This may change in
+the future, and if it does, we could then amend the extension to support
+exporting memory with a non-linear layout.
+
+=== Closing OS Handles
+
+When a call is made to `export_device_mem_handle`, the {dpcpp} implementation
+will internally create an OS specific handle to the memory region. Both CUDA and
+Level Zero allow the user to specify the type of handle to be created. However,
+this is not always respected by the Level Zero driver. For this reason, if the
+user wishes to close the OS handle returned by `export_device_mem_handle`
+without freeing the memory, they must call the appropriate OS specific API to
+close the type of handle returned by the function.
+
+When exporting a file descriptor handle on Linux, our testing has shown that the
+`close` Linux API should work.
+
+On Windows systems, the type of OS handle returned by `export_device_mem_handle`
+may not be an NT handle (e.g. it may be a KMT handle), and therefore the user
+may experience issues when trying to close the handle using the `CloseHandle`
+Windows API.
+
+The issue of closing OS handles returned by `export_device_mem_handle` is
+something we are aware of and want to address in future versions of this
+extension. Once we have a solution, we will update this specification with a
+SYCL API that will close the OS handles returned by `export_device_mem_handle`
+without freeing the memory.
+
+=== Using `sycl::malloc_device ` and `sycl::free` for exportable memory
+
+As this is an initial draft of an experimental extension, we provide explicit
+APIs for the allocation and deallocation of exportable memory. However, there
+is nothing in principle that should prevent this extensions from using
+`sycl::malloc_device` with a `sycl::property` to allocate exportable memory,
+and `sycl::free` to deallocate it. While the implementation of this in {dpcpp}
+would involve minor overhead, it would allow the user to use the same
+allocation and deallocation APIs for both exportable and non-exportable memory.
+
+We are considering this approach for future versions of this extension, but for
+this initial draft we've have decided to provide explicit APIs to simplify the
+implementation and gather early feedback.
+
+=== Querying Supported External Memory Handle Types
+
+Currently, there is no way to query which external memory handle types are
+supported by the implementation. As this is an initial draft of an
+experimental extension intended to gather early feedback, we have not
+implemented this functionality yet. However, we are aware of this limitation
+and plan to address it in future versions of this extension.
+
+As noted above in the document, the {dpcpp} implementation currently
+supports exporting memory with the `opaque_fd` and `win32_nt_handle` handle
+types, and only on the Level Zero backend.
```

Comment:
Consistent with my other comments, I think we should remove this last paragraph about what DPC++ supports.

Reply:
Removed.